docs/porting/libraries.md

```diff
@@ -1,5 +1,304 @@
+
+### App Domains
+**Why was it discontinued?** AppDomains require runtime support and are generally quite expensive. While still implemented by CoreCLR, it’s not available in .NET Native and we don’t plan on adding this capability there.  We do not consider AppDomains a good solution for the traditional use cases.
+
+**What should you use instead?** AppDomains were used for different purposes. For code isolation, we recommend processes and/or containers. For dynamic loading of assemblies, we recommend the new [AssemblyLoadContext](http://dotnet.github.io/api/System.Runtime.Loader.AssemblyLoadContext.html) class.
+
+### Remoting
+**Why was it discontinued?** The idea of .NET remoting — which is transparent remote procedure calls — has been identified as a problematic architecture. Outside of that realm, it’s also used for cross AppDomain communication. On top of that, remoting requires runtime support and is quite heavyweight.
+
+**What should you use instead?** For communication across processes, inter-process communication (IPC) should be used, such as [Pipes](http://dotnet.github.io/api/System.IO.Pipes.html) or [Memory Mapped Files](http://dotnet.github.io/api/System.IO.MemoryMappedFiles.html). Across machines, you should use a network based solution, preferably a low-overhead plain text protocol such as HTTP.  Remote proxy generation via [Castle.Core](https://github.com/castleproject/Core) is also an option.
+
+### Binary serialization
+**Why was it discontinued?** Binary serialization requires intimate knowledge of the types because it allows to serialize object graphs, which includes private state.  After a decade of servicing, we’ve learned that binary serialization is a very costly API to maintain and a considerable burden for the types supporting it, all for questionable benefits. Thus, we made the decision that serialization should be a protocol implemented on top of the available public APIs.
+
+**What should you use instead?** Choose the serialization technology that fits your goals for formatting and footprint. Popular choices include [JSON.NET](http://www.newtonsoft.com/json), [data contract serialization](http://dotnet.github.io/api/System.Runtime.Serialization.DataContractSerializer.html), [XML serialization](http://dotnet.github.io/api/System.Xml.Serialization.XmlSerializer.html), and [protobuf-net](https://github.com/mgravell/protobuf-net).  Refer to the linked resources to learn about their benefits and choose what is appropriate for your needs.
+
+### Sandboxing
+**Why was it discontinued?** Sandboxing, i.e. relying on the runtime or the framework to constrain which resources a managed application can access, is considered a non-goal for .NET Core. Sandboxing applications and components can give a false sense of security, which is why we recommend that you do not rely on it. It also makes the implementation more complicated and often negatively affects performance of applications that don’t use sandboxing.
```

Comment:
i.e. -> that is
affects performance of applications -> affects the performance of the applications